src/snapshot/metadata.rs

```diff
@@ -93,16 +86,33 @@ impl InstanceMetadata {
     }
 }
 
+impl From<&InstanceContext> for InstanceMetadata {
+    fn from(context: &InstanceContext) -> Self {
+        Self {
+            ignore_unknown_instances: false,
+            instigating_source: None,
+            relevant_paths: Vec::new(),
+            context: context.to_owned(),
+        }
+    }
+}
```

Comment:
Will do! I'd be curious to know why it's not prefered, does it produce more boilerplate?

Reply:
It's not idiomatic; generally, implementations of `From` should be used to convert cleanly between types that are equivalent representations. Or it's used to eliminate common boilerplate for types that take ownership of another. You can see that if you read through the [implementations in the standard library](https://doc.rust-lang.org/std/convert/trait.From.html#implementors).

In this case, it would be doing neither, since `InstanceContext` and `InstanceMetadata` aren't equivalent types and `InstanceMetadata` isn't just a wrapper for `InstanceContext`.